src/cljs/events.cljs

```diff
@@ -19,8 +19,8 @@
 ;; Default event for HTTP Failure
 (rf/reg-event-db
   :http-no-on-failure
-  (fn [db fail]
+  (fn [db [_ fail]]
     (-> db
         (assoc :laedt false)
-        (assoc :error fail))))
-
+        (assoc :error fail))
+    (if (== (:status fail) 401) (set! (.-location js/window) "http://localhost:8081/api/oauth2/github") ())))
```

Comment:
Should register effect to handle the side effect of (set! ...)

Reply:
Is this really needed? The side effect would be that the SPA is terminated.

Reply:
Do you mean [`rf/reg-event-db`](https://day8.github.io/re-frame/api-re-frame.core/#reg-fx).
I can then change it this evening.

Reply:
No, I meant `rf/reg-fx` and use `rf/reg-event-fx` for the :http-no-on-failure Event, with the newly defined effect (it's a bit confusing, but effects are name fx in re-frame). https://day8.github.io/re-frame/Effects/#extensible-side-effects

Reply:
> Is this really needed? The side effect would be that the SPA is terminated.

It's not strictly necessary, but technically we don't want our events to have side effects.